app/src/main/java/com/example/next_show/MainActivity.java

```diff
@@ -94,43 +96,43 @@ private void setProfileFragment(ProfileFragment profileFragment) {
-                    try {
-                        // Get trending shows
-                        Response<List<TrendingShow>> response = traktShows.trending(1, null, Extended.FULL).execute();
-                        if (response.isSuccessful()) {
-                            List<TrendingShow> repsonseShows = response.body();
-                            for (TrendingShow trending : repsonseShows) {
-                                Log.i(TAG, "Title: " + trending.show.title);
-                                Show currentShow = new Show(trending.show.title);
-                                shows.add(currentShow);
-                            }
-
-                            // show the list
-                            Log.i(TAG, "List of show titles: " + shows.toString());
+        // Trakt wrapper
+        TraktV2 trakt = new TraktV2(getString(R.string.trakt_client_id));
+        Shows traktShows = trakt.shows();
+
+        try {
+            traktShows.trending(1, null, Extended.FULL).enqueue(new Callback<List<TrendingShow>>() {
+                @Override
+                public void onResponse(Call<List<TrendingShow>> call, Response<List<TrendingShow>> response) {
+                    if (response.isSuccessful()) {
+                        List<TrendingShow> repsonseShows = response.body();
+                        for (TrendingShow trending : repsonseShows) {
+                            Log.i(TAG, "Title: " + trending.show.title);
+                            Show currentShow = new Show(trending.show.title);
+                            shows.add(currentShow);
+                        }
+
+                        // show the list
+                        Log.i(TAG, "List of show titles: " + shows.toString());
+                    } else {
+                        if (response.code() == 401) {
+                            // authorization required, supply a valid OAuth access token
+                            Log.e(TAG, "Access token required");
                         } else {
-                            if (response.code() == 401) {
-                                // authorization required, supply a valid OAuth access token
-                                Log.e(TAG, "Access token required");
-                            } else {
-                                // the request failed for some other reason
-                                Log.e(TAG, "Look at stack trace, failed" + response.code());
-                            }
+                            // the request failed for some other reason
+                            Log.e(TAG, "Look at stack trace, failed" + response.code());
                         }
-                    } catch (Exception e) {
-                        Log.e(TAG, "Did not get to response, failed", e);
                     }
-                } catch (Exception e) {
-                    e.printStackTrace();
                 }
-            }
-        });
-        thread.start();
```

Comment:
What changes did you make to ensure this isn't run on the main thread now?

Reply:
I took it out of this thread and put the call to the url with a Callback so that it runs asynchronously in enqueue()